engine/gamesys/src/gamesys/components/comp_sound.cpp

```diff
@@ -74,11 +74,11 @@ namespace dmGameSystem
 
         SoundWorld* world = new SoundWorld();
         uint32_t comp_count = dmMath::Min(params.m_MaxComponentInstances, sound_context->m_MaxComponentCount);
-        const uint32_t MAX_INSTANCE_COUNT = 32;
-        world->m_Entries.SetCapacity(MAX_INSTANCE_COUNT);
-        world->m_Entries.SetSize(MAX_INSTANCE_COUNT);
-        world->m_EntryIndices.SetCapacity(MAX_INSTANCE_COUNT);
-        memset(&world->m_Entries.Front(), 0, MAX_INSTANCE_COUNT * sizeof(PlayEntry));
+        const uint32_t max_instances = sound_context->m_MaxSoundInstances;
+        world->m_Entries.SetCapacity(max_instances);
+        world->m_Entries.SetSize(max_instances);
+        world->m_EntryIndices.SetCapacity(max_instances);
```

Comment:
One `entry` uses one sound `instance` from sound system.
We use `sound.max_sound_instances` as count of `instances` so we should use the same number for `entries`